gradle/instrumentation-library.gradle

```diff
@@ -0,0 +1,29 @@
+ext {
+  noShadowPublish = true
```

Comment:
If we have naming convention for modules, as suggested above, we may replace this configuration with convention. `publish.gradle` should figure out itself that instrumentations should publish regular jars and only exporters and agent itself should publish fat jar.

Reply:
I don't know if it's worth adding that which is sort of magic, we've already abstracted it away to `instrumentation-library.gradle`. Or are you suggesting projects shouldn't have to do `apply from: "instrumentation-library.gradle"`?

In my experience with some large Gradle projects, I've had better luck having top level logic respond to small signals from the subprojects, for example this `apply from:`, or for publish I'd usually just apply `maven-publish` in the subprojects and `plugins.withId('maven-publish')` in the top-level. Seems to reduce top-level spaghetti from what I've seen. But let me know if the removal of the `apply from:` is actually your idea here.

Reply:
No, `apply from:` is fine, I don't object it at all.

Reply:
So hopefully this is OK, as long as this detail isn't leaked into the project gradle files themselves, it seems a little nicer to have less magic here.